python/tests/test_graphql.py

```diff
@@ -568,6 +568,241 @@ def test_rename_graph():
+            }}
+        }
+    }"""
+    result = client.query(query)
+    assert result['graph']['nodes']['list'] == [{'name': 'ben'}]
+    assert result['graph']['properties']['constant']['creationTime']['value'] is not None
+    assert result['graph']['properties']['constant']['lastOpened']['value'] is not None
+    assert result['graph']['properties']['constant']['lastUpdated']['value'] is not None
+    assert result['graph']['properties']['constant']['uiProps']['value'] == '{ "target": 6 : }'
+    assert result['graph']['properties']['constant']['isArchive']['value'] == 1
+    
+    server.stop()
+
+
+def test_save_graph():
+    g = Graph()
+    g.add_edge(1, "ben", "hamza")
+    g.add_edge(2, "haaroon", "hamza")
+    g.add_edge(3, "ben", "haaroon")
+
+    work_dir = tempfile.mkdtemp()
+    os.makedirs(os.path.join(work_dir, "shivam"), exist_ok=True)
+
+    g.save_to_file(os.path.join(work_dir, "g1"))
+    g.save_to_file(os.path.join(work_dir, "shivam", "g2"))
+    g.save_to_file(os.path.join(work_dir, "shivam", "g3"))
+
+    server = RaphtoryServer(work_dir).start()
+    client = server.get_client()
+
+    # Assert if "save" graph succeeds, when parent graph belongs to different namespace
+    query = """{
+        graph(name: "g3", namespace: "shivam") { 
+            nodes { list { name } }
+            properties { constant {
+              creationTime: get(key: "creationTime") { value }
+              lastUpdated: get(key: "lastUpdated") { value }
+              lastOpened: get(key: "lastOpened") { value }
+              uiProps: get(key: "uiProps") { value }
+              isArchive: get(key: "isArchive") { value }
+            }}
+        }
+    }"""
+    result = client.query(query)    
+    # Assertions pre-save graph
+    assert result['graph']['nodes']['list'] == [{'name': 'ben'}, {'name': 'hamza'}, {'name': 'haaroon'}]
+    assert result['graph']['properties']['constant']['creationTime'] is None
+    assert result['graph']['properties']['constant']['lastOpened'] is None
+    assert result['graph']['properties']['constant']['lastUpdated'] is None
+    assert result['graph']['properties']['constant']['uiProps'] is None
+    assert result['graph']['properties']['constant']['isArchive'] is None
+    
+    save_graph = """mutation {
+      saveGraph(
+        parentGraphName: "g1",
+        graphName: "g3",
+        graphNamespace: "shivam",
+        newGraphName: "g3",
+        props: "{ \\"target\\": 6 : }",
+        isArchive: 1,
+        graphNodes: "{ \\"ben\\": {}, \\"haaroon\\": {} }"
+      )
+    }"""
+    client.query(save_graph)
+
+    result = client.query(query)
+    # Assertions post-save graph
+    assert result['graph']['nodes']['list'] == [{'name': 'ben'}, {'name': 'haaroon'}]
+    assert result['graph']['properties']['constant']['creationTime'] is None   # Since it is an existing graph
+    assert result['graph']['properties']['constant']['lastOpened']['value'] is not None
```

Comment:
We need to double check that if every time we open the graph in the UI that we don't invalidate the graph in cache because we are saving a new lastOpened prop to disk